opensearch/src/main/java/org/opensearch/sql/opensearch/planner/physical/ADOperator.java

```diff
@@ -0,0 +1,213 @@
+import org.opensearch.sql.opensearch.client.MLClient;
+import org.opensearch.sql.planner.physical.PhysicalPlan;
+import org.opensearch.sql.planner.physical.PhysicalPlanNodeVisitor;
+
+/**
+ * AD Physical operator to call AD interface to get results for
+ * algorithm execution.
+ */
+@RequiredArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+public class ADOperator extends PhysicalPlan {
+
+  @Getter
+  private final PhysicalPlan input;
+
+  @Getter
+  private final Map<String, Literal> arguments;
+
+  @Getter
+  private final NodeClient nodeClient;
+
+  @EqualsAndHashCode.Exclude
+  private Iterator<ExprValue> iterator;
+
+  private FunctionName rcfType;
+
+  @Override
+  public void open() {
+    super.open();
+    DataFrame inputDataFrame = generateInputDataset();
+    MLAlgoParams mlAlgoParams = convertArgumentToMLParameter(arguments);
+
+    MLInput mlinput = MLInput.builder()
+            .algorithm(rcfType)
+            .parameters(mlAlgoParams)
+            .inputDataset(new DataFrameInputDataset(inputDataFrame))
+            .build();
+
+    MachineLearningNodeClient machineLearningClient =
+            MLClient.getMLClient(nodeClient);
+    MLPredictionOutput predictionResult = (MLPredictionOutput) machineLearningClient
+            .trainAndPredict(mlinput)
+            .actionGet(30, TimeUnit.SECONDS);
+    Iterator<Row> inputRowIter = inputDataFrame.iterator();
+    Iterator<Row> resultRowIter = predictionResult.getPredictionResult().iterator();
+    iterator = new Iterator<ExprValue>() {
+      @Override
+      public boolean hasNext() {
+        return inputRowIter.hasNext();
+      }
+
+      @Override
+      public ExprValue next() {
+        ImmutableMap.Builder<String, ExprValue> resultSchemaBuilder = new ImmutableMap.Builder<>();
+        resultSchemaBuilder.putAll(convertRowIntoExprValue(inputDataFrame.columnMetas(),
+                inputRowIter.next()));
+        Map<String, ExprValue> resultSchema = resultSchemaBuilder.build();
+        ImmutableMap.Builder<String, ExprValue> resultBuilder = new ImmutableMap.Builder<>();
+        resultBuilder.putAll(convertResultRowIntoExprValue(
+                predictionResult.getPredictionResult().columnMetas(),
+                resultRowIter.next(),
+                resultSchema));
+        resultBuilder.putAll(resultSchema);
+        return ExprTupleValue.fromExprValueMap(resultBuilder.build());
+      }
+    };
+  }
+
+  @Override
+  public <R, C> R accept(PhysicalPlanNodeVisitor<R, C> visitor, C context) {
+    return visitor.visitAD(this, context);
+  }
+
+  @Override
+  public boolean hasNext() {
+    return iterator.hasNext();
+  }
+
+  @Override
+  public ExprValue next() {
+    return iterator.next();
+  }
+
+  @Override
+  public List<PhysicalPlan> getChild() {
+    return Collections.singletonList(input);
+  }
+
+  protected MLAlgoParams convertArgumentToMLParameter(Map<String, Literal> arguments) {
+    if (arguments.get("time_field").getValue() == null) {
```

Comment:
See multiple places using "time_field". How about creating constants and reuse?

Reply:
Couldn't find any Constants class to use at first in this package, so didn't use constants to represent these fields. Created a constant class in the latest revision to store MLCommons related constants.